Source/Core/Core/PowerPC/JitArm64/JitArm64_RegCache.cpp

```diff
@@ -143,26 +182,40 @@ void Arm64GPRCache::FlushRegister(size_t preg, bool maintain_state)
 
 void Arm64GPRCache::FlushRegisters(BitSet32 regs, bool maintain_state)
 {
+  const auto adjacent_registers = [&](size_t index){
+    if (index + 1 >= m_guest_registers.size())
+      return false;
+
+    GuestRegInfo guest_reg_1 = GetGuestByIndex(index);
+    GuestRegInfo guest_reg_2 = GetGuestByIndex(index + 1);
+
+    if (guest_reg_1.bitsize != guest_reg_2.bitsize)
+      return false;
+
+    size_t byte_size = guest_reg_1.bitsize / 8;
+    return guest_reg_2.ppc_offset == guest_reg_1.ppc_offset + byte_size;
+  };
+
   for (size_t i = 0; i < m_guest_registers.size(); ++i)
   {
     if (regs[i])
```

Comment:
Good catch. Will have to update to use BitSet64 since we have more that 32 guest registers now.